libconfluo/src/container/cursor/json_cursors.cc

```diff
@@ -0,0 +1,33 @@
+#include "container/cursor/json_cursors.h"
+
+namespace confluo {
+
+aggregated_json_cursor::aggregated_json_cursor(std::unique_ptr<offset_cursor> o_cursor,
+                                           const data_log *dlog,
+                                           const schema_t *schema,
+                                           const parser::compiled_expression &cexpr,
+                                           size_t batch_size)
+    : json_cursor(batch_size),
+      o_cursor_(std::move(o_cursor)),
+      dlog_(dlog),
+      schema_(schema),
+      cexpr_(cexpr) {
+  init();
+}
+
+size_t aggregated_json_cursor::load_next_batch() {
+  // TODO add functionality to make into a json_string
+  size_t i = 0;
+  for (; i < current_batch_.size() && o_cursor_->has_more();
+         ++i, o_cursor_->advance()) {
+    uint64_t o = o_cursor_->get();
```

Comment:
None of this logic will be needed; only need to convert each record in record_cursor to JSON.